src/libstd/thread/mod.rs

```diff
@@ -889,7 +889,7 @@ pub fn park_timeout(dur: Duration) {
     match thread.inner.state.compare_exchange(EMPTY, PARKED, SeqCst, SeqCst) {
         Ok(_) => {}
         Err(NOTIFIED) => {
-            thread.inner.state.store(EMPTY, SeqCst);
+            thread.inner.state.swap(EMPTY, SeqCst);
```

Comment:
Maybe add a comment that why `swap` rather than `store`? It's possible that the state goes to `NOTIFIED` *just* before this line and we need to acknowledge all notifications with an `Acquire` load (at the very least).